```
=========================== short test summary info ============================
FAILED tests/components/onboarding/test_views.py::test_onboarding_core_sets_up_met[pyloop]
```
even after #68732 is already included by lastest rebase 🤨

--- a/homeassistant/components/synology_dsm/const.py
+++ b/homeassistant/components/synology_dsm/const.py
@@ -121,9 +121,11 @@ class SynologyDSMSwitchEntityDescription(
 # Binary sensors
 UPGRADE_BINARY_SENSORS: tuple[SynologyDSMBinarySensorEntityDescription, ...] = (
     SynologyDSMBinarySensorEntityDescription(
+        # Deprecated, scheduled to be removed in 2022.6 (#68664)
         api_key=SynoCoreUpgrade.API_KEY,
         key="update_available",
         name="Update Available",
+        entity_registry_enabled_default=False,
         device_class=BinarySensorDeviceClass.UPDATE,
         entity_category=EntityCategory.DIAGNOSTIC,
     ),

--- a/homeassistant/components/synology_dsm/update.py
+++ b/homeassistant/components/synology_dsm/update.py
@@ -1,7 +1,7 @@
 """Support for Synology DSM update platform."""
 from __future__ import annotations
 
-from typing import Any, Final
+from typing import Final
 
 from synology_dsm.api.core.upgrade import SynoCoreUpgrade
 
@@ -10,7 +10,6 @@ from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity import EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
 from . import SynoApi, SynologyDSMBaseEntity
 from .const import (
@@ -50,15 +49,6 @@ class SynoDSMUpdateEntity(SynologyDSMBaseEntity, UpdateEntity):
     """Mixin for update entity specific attributes."""
 
     entity_description: SynologyDSMUpdateEntityEntityDescription
-
-    def __init__(
-        self,
-        api: SynoApi,
-        coordinator: DataUpdateCoordinator[dict[str, dict[str, Any]]],
-        description: SynologyDSMUpdateEntityEntityDescription,
-    ) -> None:
-        """Initialize the Synology DSM binary_sensor entity."""
-        super().__init__(api, coordinator, description)
 
     @property
     def current_version(self) -> str | None:

--- a/homeassistant/components/synology_dsm/update.py
+++ b/homeassistant/components/synology_dsm/update.py
@@ -4,6 +4,7 @@ from __future__ import annotations
 from typing import Final
 
 from synology_dsm.api.core.upgrade import SynoCoreUpgrade
+from yarl import URL
 
 from homeassistant.components.update import UpdateEntity
 from homeassistant.config_entries import ConfigEntry
@@ -65,10 +66,14 @@ class SynoDSMUpdateEntity(SynologyDSMBaseEntity, UpdateEntity):
     @property
     def release_url(self) -> str | None:
         """URL to the full release notes of the latest version available."""
-        base_url = "http://update.synology.com/autoupdate/whatsnew.php?"
-        if (details := self._api.upgrade.available_version_details) is not None:
-            url = f"{base_url}model={self._api.information.model}&update_version={details['buildnumber']}"
-            if details.get("nano") > 0:
-                return f"{url}-{details['nano']}"
-            return url
-        return None
+        if (details := self._api.upgrade.available_version_details) is None:
+            return None
+
+        url = URL("http://update.synology.com/autoupdate/whatsnew.php")
+        query = {"model": self._api.information.model}
+        if details.get("nano") > 0:
+            query["update_version"] = f"{details['buildnumber']}-{details['nano']}"
+        else:
+            query["update_version"] = details["buildnumber"]
+
+        return url.update_query(query).human_repr()

